core/controlplane/config/templates/cluster.yaml

```diff
@@ -491,14 +485,8 @@ worker:
 #      ephemeralImageStorage:
 #        enabled: true
 #
-#      # When enabled this will install kiam using the repo from `kiamImage` (ref https://github.com/uswitch/kiam)
-#      # It will also grant sts:assumeRole permission to the IAM role for worker nodes in this pool.
-#      # This will use SSL certificates generated during `kube-aws render credentials`, ensure this has been run with kube-aws v0.9.9+
-#      kiamSupport:
-#        enabled: true
-#
-#      # When enabled it will grant sts:assumeRole permission to the IAM roles for worker nodes in this pool.
-#      # This is intended to be used in combination with managedIamRoleName. See #297 for more information.
+#      # When enabled this will grant sts:assumeRole permission to the IAM roles for worker nodes in this pool.
+#      # This is intended to be used in combination with nodePools[].iam.role.name. See #297 for more information.
 #      kube2IamSupport:
 #        enabled: true
```

Comment:
@mumoshu do you know why we have `kube2IamSupport` on node pools? I thought it was to allow only some node pools to have this enabled but having looked at it again I cannot see anything which prevents the daemon set from running across everything.

Reply:
@c-knowles Wow, good catch! I can't imagine how I miss that until now, but probably we need node affinity:

```yaml
              affinity:
                nodeAffinity:
                  requiredDuringSchedulingIgnoredDuringExecution:
                    nodeSelectorTerms:
                    - matchExpressions:
                      - key: "kube-aws.coreos.com/assume-role-supported"
                        operator: "In"
                        values:
                        - "true"
```

and corresponding node labels?

Reply:
@mumoshu yeah something like that, it seems totally missing from the beginning of this feature as far as I can tell.